src/components/molecules/denomination-grid/index.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useEffect, useMemo, useRef, useState } from "react"
+import { useObserveWidth } from "../../../hooks/use-observe-width"
+import DenominationBadge from "../../atoms/denomination-badge"
+
+export type GiftCardVariant = {
+  prices: {
+    currency_code: string
+    amount: number
+  }[]
+}
+
+type DenominationGridProps = {
+  variants: GiftCardVariant[]
+  defaultCurrency: string
+}
+
+const DenominationGrid: React.FC<DenominationGridProps> = ({
+  variants,
+  defaultCurrency,
+}) => {
+  const containerRef = useRef(null)
+
+  const width = useObserveWidth(containerRef)
+
+  const [columns, setColumns] = useState(7)
+
+  // We estimate that a Badge is rouglhy 70px and use this to calulate how many we can fit in a row
+  // It is a bit of a "magic" number, but it works for now.
+  useEffect(() => {
+    if (width) {
+      setColumns(Math.floor(width / 70) - 1)
```

Comment:
In the case where width < 70, columns will be set to -1, which will make the denominations behave in a funny way. You can try it by resizing the viewport to a smaller size (around < 330px)

One nifty solution would be to do this instead: `setColumns(Math.max(Math.floor(width/70) - 1, 1)`. This would basically say: "always show at least 1 denomination".